hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/node/NewNodeHandler.java

```diff
@@ -41,6 +46,11 @@ public NewNodeHandler(PipelineManager pipelineManager,
   @Override
   public void onMessage(DatanodeDetails datanodeDetails,
       EventPublisher publisher) {
-    pipelineManager.triggerPipelineCreation();
+    try {
+      pipelineManager.triggerPipelineCreation();
+    } catch (NotLeaderException ex) {
+      LOG.warn("Not the current leader SCM and cannot start pipeline" +
```

Comment:
Can we put this as debug log to avoid spamming the SCM log?

Reply:
Updated